packages/trace-viewer/src/ui/uiModeView.tsx

```diff
@@ -354,14 +354,26 @@ export const UIModeView: React.FC<{}> = ({
     return () => disposable.dispose();
   }, [runTests, testServerConnection, watchAll, watchedTreeIds, teleSuiteUpdater, projectFilters]);
 
+  const hideOutput = React.useCallback(() => setIsShowingOutput(false), [setIsShowingOutput]);
+  const showOutput = React.useCallback(() => {
+    setIsShowingOutput(true);
+    setOutputContainsError(false);
+  }, [setIsShowingOutput, setOutputContainsError]);
+  const toggleOutput = React.useCallback(() => {
+    if (isShowingOutput)
+      hideOutput();
+    else
+      showOutput();
+  }, [isShowingOutput, showOutput, hideOutput]);
```

Comment:
```suggestion
  const toggleOutput = React.useCallback(
      () =>
        setIsShowingOutput(isShowingOutput => {
          if (!isShowingOutput)
            setOutputContainsError(false);
          return !isShowingOutput;
        }),
      [],
  );
```

You never want to rely on the "current" rendered value of React state in a callback. Doing so can result in the callback performing on stale state, resulting in unexpected consequences.

Reply:
This removes the data dependency that `useEffect` has, so that event listener is not recreated unnecessarily.